cli/java/com/engflow/bazel/invocation/analyzer/consoleoutput/ConsoleOutput.java

```diff
@@ -245,16 +264,49 @@ String formatAnalysisOutput(SuggestionOutput suggestionOutput) {
               .collect(Collectors.toList());
       addSection(sb, HEADING_CAVEATS, formattedCaveats, ConsoleOutputStyle.TEXT_YELLOW);
     }
+    return sb.toString();
+  }
 
-    if (!suggestionOutput.getCaveatList().isEmpty()) {
-      List<String> formattedCaveats =
-          suggestionOutput.getCaveatList().stream()
-              .map(caveat -> formatCaveat(caveat))
-              .collect(Collectors.toList());
-      sb.append(NEWLINE);
-      addSection(sb, HEADING_CAVEATS, formattedCaveats, ConsoleOutputStyle.TEXT_YELLOW);
+  @VisibleForTesting
+  String formatSuggestionOutputCaveats(List<Caveat> caveats) {
+    if (caveats.isEmpty()) {
+      return "";
     }
+    StringBuilder sb = new StringBuilder();
+    addSection(
+        sb,
+        HEADING_SUGGESTION_OUTPUT_CAVEATS,
+        caveats.stream().map(caveat -> formatCaveat(caveat)).collect(Collectors.toList()),
+        ConsoleOutputStyle.TEXT_YELLOW);
+    return sb.toString();
+  }
 
+  @VisibleForTesting
+  String formatFailures(List<SuggestionOutput.Failure> failures) {
+    if (failures.isEmpty()) {
+      return "";
+    }
+    StringBuilder sb = new StringBuilder();
+    for (SuggestionOutput.Failure failure : failures) {
```

Comment:
Why are we adding newlines based on the number of failures here? This looks left over from refactoring?

Reply:
Removed.